dapps/shop/backend/queues/makeOfferProcessor.js

```diff
@@ -0,0 +1,127 @@
+const queues = require('./queues')
+
+const Web3 = require('web3')
+
+const { ListingID } = require('../utils/id')
+const { Shop, Network } = require('../models')
+const { post, getBytes32FromIpfsHash } = require('../utils/_ipfs')
+const encConf = require('../utils/encryptedConfig')
+const abi = require('../utils/_abi')
+
+const ZeroAddress = '0x0000000000000000000000000000000000000000'
+
+function attachToQueue() {
+  const queue = queues['makeOfferQueue']
+  queue.process(processor)
+  queue.resume() // Start if paused
+}
+
+/**
+ * Processes a credit card transaction and submit it to the blockchain.
+ *
+ * job.data should have {shopId, amount, encryptedData}
+ * @param {*} job
+ */
+async function processor(job) {
+  job.log('Started processing')
+  const { shopId, amount, encryptedData } = job.data
+  const shop = await getShop(shopId)
+  job.log('Load encrypted shop config')
+  const shopConfig = await getShopConfig(shop)
```

Comment:
I don't think this needs to be async

Reply:
Both getting the shop from the db, and the shop config over a web request are async.

Reply:
There's nothing async inside the `getShopConfig` function though... do it doesn't need to be async from what I can see

Reply:
Ah! I see, I was looking up the wrong get config when creating this method. I've removed the async from this one, since it was just decrypting the config it already had from the database.